worker/worker.go

```diff
@@ -60,7 +60,21 @@ func (ts TaskState) String() string {
 }
 
 func (w *Worker) Join(serverEndpoint string) {
-	httpResponse := utils.SignedPost(w.Id, w, serverEndpoint + "/workers")
+	headers := http.Header{}
+	publicKey := utils.GetPublicKey(w.Id)
+	parsedKey, err := json.Marshal(publicKey)
+
+	if err != nil {
+		log.Fatal("error on marshalling public key")
+	}
+
+	headers.Set("PUBLIC-KEY", string(parsedKey))
+	httpResponse, err := utils.Post(w.Id, w, http.Header{}, serverEndpoint + "/workers")
```

Comment:
You are creating a header but you are passing an empty header when calling the Post method.

Reply:
nice catch